src/test/components/elements/EventBookingForm.test.tsx

```diff
@@ -0,0 +1,193 @@
+    options.forEach((option, index) => {
+      expect(option.value).toBe(yearGroups[index]);
+    });
+  });
+
+  it("if the event is student only, the event booking details card displays a warning message", () => {
+    const event = augmentEvent(mockEvent);
+    setupTest({ role: "STUDENT", user: mockUser, event: { ...event, isStudentOnly: true } });
+    const warningMessage = screen.getByTestId("student-only");
+    expect(warningMessage).toBeInTheDocument();
+  });
+
+  it("if user is not a student, Job title is requested", () => {
+    setupTest({ role: "TEACHER", user: mockUser });
+    const jobTitle = screen.getByLabelText("Job title");
+    expect(jobTitle).toBeInTheDocument();
+    fireEvent.change(jobTitle, { target: { value: "Head of CS" } });
+    expect(updateAdditionalInformation).toHaveBeenCalledWith({ jobTitle: "Head of CS" });
+  });
+
+  it("if user is not a student, level of teaching experience is requested", () => {
+    setupTest({ role: "TEACHER", user: mockUser });
+    const teachingExperience = screen.getByLabelText("Level of teaching experience");
+    expect(teachingExperience).toBeInTheDocument();
+    fireEvent.change(teachingExperience, { target: { value: "2 years" } });
+    expect(updateAdditionalInformation).toHaveBeenCalledWith({ experienceLevel: "2 years" });
+  });
+
+  it("if event is not virtual, dietary or medical requirements are requested", () => {
+    const event = augmentEvent(mockEvent);
+    setupTest({ role: "STUDENT", user: mockUser, event: { ...event, isVirtual: false } });
+    const dietaryOrMedicalRequirements = screen.getByLabelText(/dietary requirements/i);
+    expect(dietaryOrMedicalRequirements).toBeInTheDocument();
+    fireEvent.change(dietaryOrMedicalRequirements, { target: { value: "Vegan" } });
+    expect(updateAdditionalInformation).toHaveBeenCalledWith({ medicalRequirements: "Vegan" });
+  });
+
+  it("if event is not virtual, accessibility requirements are requested", () => {
+    const event = augmentEvent(mockEvent);
+    setupTest({ role: "STUDENT", user: mockUser, event: { ...event, isVirtual: false } });
+    const accessibilityRequirements = screen.getByLabelText(/accessibility requirements/i);
+    expect(accessibilityRequirements).toBeInTheDocument();
+    fireEvent.change(accessibilityRequirements, { target: { value: "Wheelchair access" } });
+    expect(updateAdditionalInformation).toHaveBeenCalledWith({ accessibilityRequirements: "Wheelchair access" });
+  });
+
+  it("if user is not a teacher and event is not virtual, emergency contact name and telephone number is requested", async () => {
+    const event = augmentEvent(mockEvent);
+    setupTest({ role: "STUDENT", user: mockUser, event: { ...event, isVirtual: false } });
+    const emergencyContactDetails = screen.getByRole("heading", { name: /emergency contact details/i });
+    expect(emergencyContactDetails).toBeInTheDocument();
+    const contactDetails = [
```

Comment:
I think this is essentially a parametrised test, so you could rewrite this as such.

Reply:
There is a native way to do parametrised tests in Jest: https://jestjs.io/docs/api#testeachtablename-fn-timeout

We already have some examples of this in the codebase, you can check those as an example.

Also, I just noticed that the second test (`it("has first name, last name, email address...`) should also be parametrised.

Reply:
Yeah I I know about it.each and have used it elsewhere in the tests, but I was having difficulty with customising the description of the test, since the only thing I want to pass into the description of the test is the "testCase" of `contact name` or `telephone number` but neither of those are required for the actual test, just for the test description. 

I did also consider parameterising the other test you mention, however it would require repeatedly setting up the test environment for each field, so I thought it would be more efficient to do a single test that checks for the 6 fields - if it fails you can clearly see which one it hasn't found?

Reply:
Got it working now and parameterised the other test too.

Reply:
> it would be more efficient

That's fair, I didn't take into account that setting up these tests is a costly operation. I see that you've changed now, but your original solution was also fine, so apologies.

> if it fails you can clearly see which one it hasn't found

You can only clearly see the failures with the parametrised tests. With the manual for loops you can't. For instance, imagine that items number 2 and 3 fail. You'll see both of them failing with parametrised tests, but you'll only see number 2 failing with a manual for loop, as the test will stop after encountering a failure.

Reply:
In fairness the tests still seem to run fast enough here as it's a very simple one so I'll leave it parameterised for clarity.